sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/maskExpressions.scala

```diff
@@ -223,16 +223,32 @@ case class Mask(
     val fifthGen = children(4).genCode(ctx)
     val resultCode =
       f(firstGen.value, secondGen.value, thirdGen.value, fourthGen.value, fifthGen.value)
-    ev.copy(
-      code = code"""
+    if (nullable) {
```

Comment:
@bersprockets Have you tried using the `NullIntolerant` trait on the expression `Mask` instead?

Reply:
@gengliangwang With this function, only the first parameter affects whether NULL is returned. For the other parameters, NULL input has special meaning. An example from the doc:
```
      > SELECT _FUNC_('AbCD123-@$#', NULL, NULL, NULL, 'o');
        AbCD123oooo
```
Mixing in `NullIntolerant` causes the above query to return NULL:
```
spark-sql> SELECT MASK('AbCD123-@$#', NULL, NULL, NULL, 'o');
NULL
Time taken: 0.063 seconds, Fetched 1 row(s)
```

Reply:
I see. Thanks for the explaination!